src/blueprint/from_config/operators/expr.rs

```diff
@@ -0,0 +1,40 @@
+use crate::blueprint::*;
+use crate::config;
+use crate::config::{Config, ExprNode, Field};
+use crate::try_fold::TryFold;
+use crate::valid::Valid;
+
+pub fn update_expr(
+  operation_type: &config::GraphQLOperationType,
+) -> TryFold<'_, (&Config, &Field, &config::Type, &str), FieldDefinition, String> {
+  TryFold::<(&Config, &Field, &config::Type, &str), FieldDefinition, String>::new(
+    |(config, field, ty, name), b_field| {
+      let Some(expr) = &field.expr else {
+        return Valid::succeed(b_field);
+      };
+
+      match &expr.body {
+        ExprNode::Http(http) => {
```

Comment:
Should leverage the `operators::http` to compile `config::Http` to `expression::Http`

Reply:
I wasn't quite sure what you meant here since I have used `update_http` etc. from `operators::http`. I've refactored the code to split the match and the operator application, is this closer to what you had in mind?

Reply:
It gets quite clunky when compiling basic expressions like if, I've tried to implement a sample in #862